src/test/java/cloud/fogbow/ras/core/plugins/interoperability/opennebula/publicip/v5_4/OpenNebulaPublicIpPluginTest.java

```diff
@@ -139,296 +128,463 @@ public void testHasFailedUnsuccessful() {
-	public void testGetInstanceSuccessfully() throws FogbowException {
+	public void testAttachPublicIpToComputeFail() throws UnauthorizedRequestException, InstanceNotFoundException, InvalidParameterException {
 		// set up
-		Client client = Mockito.mock(Client.class);
-		PowerMockito.mockStatic(OpenNebulaClientUtil.class);
-		BDDMockito.given(OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString()))
-				.willReturn(client);
-
-		PublicIpOrder publicIpOrder = createPublicIpOrder();
-		String virtualMachineId = publicIpOrder.getComputeId();
-		VirtualMachine virtualMachine = Mockito.mock(VirtualMachine.class);
-		BDDMockito.given(OpenNebulaClientUtil.getVirtualMachine(Mockito.eq(client), Mockito.eq(virtualMachineId)))
-				.willReturn(virtualMachine);
-		Mockito.when(virtualMachine.xpath(OpenNebulaPuplicIpPlugin.EXPRESSION_IP_FROM_NETWORK))
-				.thenReturn(FAKE_IP_ADDRESS);
+		String template = this.getNicTemplate();
+		String message = String.format(Messages.Error.ERROR_WHILE_CREATING_NIC, template) + " " +
+				String.format(Messages.Error.ERROR_MESSAGE, this.response.getMessage());
 
-		CloudUser cloudUser = createCloudUser();
-		PublicIpInstance expected = createPublicIpInstance();
+		Mockito.doReturn(template).when(this.plugin).createNicTemplate(Mockito.anyString());
+		Mockito.when(this.virtualMachine.nicAttach(Mockito.anyString())).thenReturn(this.response);
+		Mockito.when(this.response.isError()).thenReturn(true);
 
 		// exercise
-		PublicIpInstance instance = this.plugin.getInstance(publicIpOrder, cloudUser);
+		try {
+			this.plugin.attachPublicIpToCompute(this.client, this.instanceId, this.computeId);
+			Assert.fail();
+		} catch (InvalidParameterException e) {
+			Assert.assertEquals(message, e.getMessage());
+		}
 
 		// verify
-		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, VerificationModeFactory.times(1));
-		OpenNebulaClientUtil.createClient(Mockito.anyString(), Mockito.anyString());
+		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, Mockito.times(TestUtils.RUN_ONCE));
+		OpenNebulaClientUtil.getVirtualMachine(Mockito.eq(this.client), Mockito.eq(this.computeId));
+
+		Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).createNicTemplate(Mockito.eq(this.instanceId));
+		Mockito.verify(this.virtualMachine, Mockito.times(TestUtils.RUN_ONCE)).nicAttach(Mockito.eq(template));
+		Mockito.verify(this.response, Mockito.times(TestUtils.RUN_ONCE)).isError();
+	}
 
-		PowerMockito.verifyStatic(OpenNebulaClientUtil.class, VerificationModeFactory.times(1));
-		OpenNebulaClientUtil.getVirtualMachine(Mockito.eq(client), Mockito.eq(virtualMachineId));
+	// test case: when invoking deleteInstance with valid order and cloud user, the plugin
+	// should retrieve the respective ONe virtual machine, shut it off, detach its public ip nic,
+	// delete the fogbow security group and finally delete the public ip instance network reserve
+	@Test
+	public void testDeleteInstance() throws FogbowException {
```

Comment:
This test is failing...
Please check and correct.

Reply:
done